MacOS Build Error: Qt 5 and 6 Simultaneously Installed.

This error seems to appear when `brew install qt@5` is installed at the same time as: `brew install qt`. (ie, qt version 6):

``` bash
% brew list --version | grep qt

qt 6.3.1_1
qt@5 5.15.5_1
```


```
Making all in src
  CXX      qt/libbitcoinqt_a-bantablemodel.o
In file included from qt/bantablemodel.cpp:14:
In file included from /usr/local/Cellar/qt@5/5.15.5_1/lib/QtCore.framework/Headers/QLocale:1:
/usr/local/Cellar/qt@5/5.15.5_1/lib/QtCore.framework/Headers/qlocale.h:1028:25: error: unknown type name 'QStringRef'; did you mean 'CFStringRef'?
    short toShort(const QStringRef &s, bool *ok = nullptr) const;
                        ^~~~~~~~~~
                        CFStringRef
/usr/local/include/QtCore/qdebug.h:531:1: note: 'CFStringRef' declared here
Q_FORWARD_DECLARE_CF_TYPE(CFString);
^
/usr/local/include/QtCore/qglobal.h:802:77: note: expanded from macro 'Q_FORWARD_DECLARE_CF_TYPE'
#  define Q_FORWARD_DECLARE_CF_TYPE(type) typedef const struct __ ## type * type ## Ref
                                                                            ^
<scratch space>:112:1: note: expanded from here
CFStringRef
^
```

```bash
% git clean -ffxd
% git rev-parse HEAD
e191fac4f3c37820f0618f72f0a8e8b524531ab8
% ./autogen.sh
% ./configure
% make
```
output: https://gist.github.com/da2ce7/c697259e204c961876ad7f7057e95b34

config.log: https://gist.github.com/da2ce7/ec7276de7d0284681cd73b0cdcb90f3b

brew list --version --formulae: https://gist.github.com/da2ce7/25d6473224a1d69a28cf24e44abb55b7

_Originally posted by @da2ce7 in https://github.com/bitcoin/bitcoin/issues/21901#issuecomment-1229561087_

--- a/doc/build-osx.md
+++ b/doc/build-osx.md
@@ -96,14 +96,6 @@ Skip if you don't intend to use the GUI.
 brew install qt@5
 ```
 
-Ensure that the `qt@5` package is installed, not the `qt` package.
-If 'qt' is installed, the build process will fail.
-if installed, remove the `qt` package with the following command:
-
-``` bash
-brew uninstall qt
-```
-
 Note: Building with Qt binaries downloaded from the Qt website is not officially supported.
 See the notes in [#7714](https://github.com/bitcoin/bitcoin/issues/7714).
 

